tests/unit/test_glancesync_serversfacade.py

```diff
@@ -29,24 +29,193 @@
 import tempfile
 import unittest
 import copy
-
+from mock import patch, MagicMock, call, ANY
 from keystoneclient.auth.identity import v2, v3
+from multiprocessing import TimeoutError
 
-from glancesync.glancesync_serversfacade import ServersFacade
+from glancesync.glancesync_serversfacade import ServersFacade,\
+    GlanceFacadeException
 from glancesync.glancesync_image import GlanceSyncImage
 from glancesync.glancesync_region import GlanceSyncRegion
 
-
-"""This is an integration test to verify that the facade works correctly
+"""This environment variable activates a pair of
+integration test to verify that the facade works correctly
 using a real server.
 
-Don't activate this test unless you know what are you doing"""
+Don't activate these tests unless you know what are you doing"""
 
 testingFacadeReal = False
 if 'TESTING_FACADE' in env:
     testingFacadeReal = True
 
 
+class MyOpenStackClients(MagicMock):
+    def get_session(self):
+        return ""
+
+    def get_regions(self, service):
+        return ['fakeregion', 'fakeregion2']
+
+    def get_tenant_id(self):
+        return 'tenantid1'
+
+mock_osclients = MyOpenStackClients()
+
+class TestGlanceServersFacadeM(unittest.TestCase):
+    """Test the facade using a mock, that is, only checks that the right calls
+    to osclients are made"""
+    @patch('glancesync.glancesync_serversfacade.OpenStackClients', mock_osclients)
+    def setUp(self):
+        target = dict()
+        target['target_name'] = 'master'
+        target['user'] = 'fakeuser'
+        target['password'] = 'fakepassword'
+        target['keystone_url'] = 'http://127.0.0.1/'
+        target['tenant'] = 'faketenant'
+        target['use_keystone_v3'] = False
+        self.target = target
+
+        self.region = 'fakeregion'
+        targets = dict()
+        targets['master'] = target
+        self.region_obj = GlanceSyncRegion(self.region, targets)
+
+        mock_osclients.reset_mock()
+        self.facade = ServersFacade(self.target)
+        self.facade.images_dir = None
+
+
+        image = GlanceSyncImage('imagetest', '01', self.region, None, False)
+        image.raw = dict()
+        image.raw['disk_format'] = 'qcow2'
+        image.raw['is_public'] = False
+        image.raw['protected'] = False
+        image.raw['container_format'] = 'bare'
+        image.raw['min_ram'] = '0'
+        image.raw['min_disk'] = '0'
+
+        self.image = image
+
+    def tearDown(self):
+        if self.facade.images_dir:
+            os.unlink(self.facade.images_dir + '/01')
+            os.rmdir(self.facade.images_dir)
+
+    def test_gettenantid(self):
```

Comment:
A brief description of what it is testing is welcome (the same for the rest of test cases)